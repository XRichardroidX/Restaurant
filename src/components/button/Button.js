import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ ...props }) {
  return (
    <button
      id={props.id}
      className={`rounded min-w-fit capitalize ${props.twClasses}`}
      type={props.type === 'submit' ? 'submit' : 'button'}
      onClick={props.handleClick}
    >
      {props.text}
    </button>
  );
}

Button.defaultProps = {
  type: 'button',
  text: '',
  handleClick: () => {},
  twClasses: '',
};

Button.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  text: PropTypes.string,
  twClasses: PropTypes.string,
  handleClick: PropTypes.func,
};
