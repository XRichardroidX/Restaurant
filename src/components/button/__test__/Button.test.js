import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer
      .create(<Button id="button-id" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the given classNames', () => {
    render(
      <Button id="button-id" twClasses="bg-red-500" />,
    );

    const button = screen.getByRole('button');
    const { classList } = button;

    expect(classList).toContain('bg-red-500');
  });

  it('renders with the correct text', () => {
    render(<Button id="button-id" text="My-Button" />);

    const button = screen.getByRole('button');

    expect(button.textContent).toContain('My-Button');
  });
});
