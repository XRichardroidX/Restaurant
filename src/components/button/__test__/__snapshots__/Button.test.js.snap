// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly in the DOM 1`] = `
<button
  className="rounded min-w-fit capitalize "
  id="button-id"
  onClick={[Function]}
  type="button"
/>
`;
